### PR TITLE
Electrified animals

### PR DESCRIPTION
**Pull request by Fox-McCloud**
## About The Pull Request

Makes a few tweaks to electrocution, so all living mobs can now be shocked. `electrocute_act` was changed, a while back, so that it works for simple mobs. Unfortunately, in this process, no one updated electrocution code to allow for simple mobs to be shocked by various sources (like the Tesla).

Additional tweaks/changes:

- Hallucinating a shock now uses `electrocute_act`, but is an illusion. This makes it more convincing as it deals stamina damage and will not go through if you're wearing insulated gloves.

- Fixes `COMSIG_LIVING_ELECTROCUTE_ACT` not being sent if the mob was a carbon or above, thus not resulting in nanite levels dropping

  - `electrocute_act` is completely overridden for carbon and above (for good reason, to be honest). There's never a parent call for `carbon/electrocute_act` (there is for human, however). As such, carbon needs to send this signal for certain nanites to work.

- Also fixes ethereal blood not bypassing resists (as with Teslium).

- Lastly, ensures `electrocute_act` has consistent arguments, across all procs---this has been an ongoing and major issue with this proc for a long time; now should be corrected.

## Why It's Good For The Game

There's a base underlying system for electrocution that is now nearly universal. As such, it makes sense that you should be able to shock simple mobs. This adds an additional level of potential gameplay to existing shocking mechanics and ensures consistency of behavior, for shocking, across all mobs.

Hallucinating shocks on doors being altered to an illusion makes it more convincing, which makes the hallucination more likely to trick players, which is a good thing.

## Changelog
:cl: Fox McCloud
add: Simple mobs can now be electrocuted/shocked by various sources, including the tesla
fix: Fixes not losing nanites upon being shocked
tweak: Hallucinating getting shocked on a door should be slightly more convincing
/:cl:

### CODE REVIEW

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -110,19 +110,14 @@
 		if(2)
 			if(scooldown < world.time)
 				if(M.health >= 0)
-					if(ishuman(M)||ismonkey(M))
+					if(isliving(M) && !iscyborg(M))
 						M.electrocute_act(5, "[user]", safety = 1)
```
Comment (Fox-McCloud) [inline code/game/objects/items/robot/robot_items.dm]: `TRUE`

**Comment by Angustmeta** [comment]
I'm glad to see simplemobs becoming less simple, and the shocks not removing nanites being fixed. 

But does wearing insulated gloves now protect you from nanites making you hallucinate being shocked? Considering there are ways to get shocked even with insulated, such as teslium, wiz spells (tesla blast) or nanites (I think),  I wouldn't want this to break the shock hallucination nanites which is currently a fun way to stun people.

**Comment by Barhandar** [comment]
Do silicons dream of electrified sheep?

**Comment by XDTM** [comment]
Is the illusion shock visible to bystanders?

**Comment by IndieanaJones** [comment]
So, does this mean tasers and stuff now work on simplemobs?

**Comment by Fox-McCloud** [comment]
>Is the illusion shock visible to bystanders?

It would be, yes, though I consider that a fairly small price to pay for making it more compelling to the hallucinatee.

**Comment by nicbn** [comment]
Hallucination shouldn't be visible to bystanders, that's the whole point of it being and hallucination

**Comment by Fox-McCloud** [comment]
Need to be informed of what, if anything, is to be done with this PR.

**Comment by optimumtact** [comment]
do you know a way to make maintainers active when they're burned out due to community criticism and toxicity?

**Comment by Dorsisdwarf** [comment]
Does this mean that at a future date silicons could be made to gain charge on being electrified?

**Comment by Fox-McCloud** [comment]
>do you know a way to make maintainers active when they're burned out due to community criticism and toxicity?

Man, If you find a solution to this, it'll help my side of the equation too. Maybe we should put our heads together and come up with a solution; we'll be rich.

>Does this mean that at a future date silicons could be made to gain charge on being electrified?

This was technically already possible--it's just that no one has done it yet (also it would be pretty abuseable, since borgs can shock doors willy nilly).

**Review by AnturK — DISMISSED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
> Hallucination shouldn't be visible to bystanders, that's the whole point of it being and hallucination

**Comment by AnturK** [comment]
Since you already have the illusion argument just modify the message/animation ?

**Comment by Fox-McCloud** [comment]
>Since you already have the illusion argument just modify the message/animation

Yeah, I an do that. Just been busy. Will try to update this when I can.

**Comment by optimumtact** [comment]
remember when simple mobs were simple

**Pull request closed without merging**